kube-config-files/binary-ingester-service.yaml

```diff
@@ -0,0 +1,104 @@
+###############################################################################
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: binary-ingester
+  labels:
+    app: binary-ingester
+    visualize: "true"
+spec:
+  replicas: 2
+  selector:
+    matchLabels:
+      app: binary-ingester
+  template:
+    metadata:
+      labels:
+        app: binary-ingester
+        visualize: "true"
+    spec:
+      containers:
+      - name: binary-ingester
+        image: up-registry.ft.com/coco/ingester:v65
+        env:
+        - name: JAVA_OPTS
+          value: "-Xms256m -Xmx256m -XX:+UseG1GC -server"
+        - name: APP_PORT
+          value: "8080"
+        - name: ADMIN_PORT
+          value: "8080"
+        - name: KAFKA_HOST
+          value: "kafka"
+        - name: KAFKA_PORT
+          value: "9092"
+        - name: KAFKA_GROUPNAME
+          value: "BinaryIngester"
+        - name: KAFKA_SYSTEM_ID
+          value: "BinaryIngester"
+        - name: KAFKA_TOPIC_NAME
+          value: "CmsPublicationEvents"
+        - name: TRANSFORMER_PROXY
+          value: "localhost:8080"
+        - name: WHITELIST
+          value: "^http://methode-image-binary-mapper(-pr|-iw)?(-uk-.*)?\\.svc\\.ft\\.com(:\\d{2,5})?/image/binary/[\\w-]+.*$" 
+        - name: WRITER_HEADER
+          value: "binary-writer"
+        - name: WRITER_NAME
+          value: "BinaryWriter"
+        - name: USE_GZIP_FOR_REQUESTS
+          value: "false"
+        - name: WRITER_HEALTHCHECK_NAME
+          value: "Can connect to the Binary Writer"
+        - name: WRITER_HEALTHCHECK_TECH_SUMMARY
+          value: "Tests that the build-info endpoint for the Binary Writer returns a 200 response"
+        - name: MESSAGE_FORWARDING_ENABLED
+          value: "false"
+        - name: FORWARD_TO_TOPIC_NAME
+          value: "PostPublicationEvents"
+        - name: KAFKA_PROXY
+          value: "kafka:8082"
+        - name: VULCAN_HOST
+          value: "binary-writer:8080"
+        - name: ZOOKEEPER_HOST
+          value: "kafka"
+        - name: ZOOKEEPER_PORT
+          value: "2181"
+        - name: UPP_GATEWAY_PATH
+          value: ""
+        ports:
+        - containerPort: 8080
+        livenessProbe:
```

Comment:
Liveness probe looks very good.
We should also add a readiness probe for the containers that have the health check enabled. The readiness probe should make an http GET call to the __gtg endpoint. The __gtg is a summary of the health check, it will return 200 if all the check passes, and 500 if any check fails.